sdk/lib/crt/misc/misc.cmake

```diff
@@ -0,0 +1,29 @@
+
+list(APPEND LIBCNTPT_MISC_SOURCE
+    misc/fltused.c
+)
+
+if(ARCH STREQUAL "i386")
+    list(APPEND LIBCNTPR_ASM_SOURCE
+        misc/i386/readcr4.S
+    )
+endif()
+
+list(APPEND CRT_MISC_SOURCE
+    ${LIBCNTPT_MISC_SOURCE}
+    misc/__crt_MessageBoxA.c
+    misc/amsg.c
+    misc/assert.c
+    misc/environ.c
+    misc/getargs.c
+    misc/i10output.c
+    misc/initterm.c
+    misc/lock.c
+    misc/purecall.c
+    misc/stubs.c
+    misc/tls.c
+)
+
+add_library(getopt misc/getopt.c)
```

Comment:
Wouldn't it make sense to keep this add_library into the "root" crt cmake file? Since somehow this is a separate module.

Reply:
Eventually I would like to replace these file collections with object libraries to avoid compiling the exact same stuff twice. That would put stuff like this into the folder where it resides in. I didn't want to do that right away, because we still use CMake defines to compille code differently between libcntpr and crt in a number of places, so this needs to be fixed first. And stuff like getopt and atan2 are IMO "hack- libraries" rather than real libraries.

Reply:
It is sometimes necessary to recompile things "twice" or more because they get compiled with different defines enabled, e.g. the win32 CRT versus the NT user-mode (or even kernel-mode) CRT.

Reply:
I agree, but in those cases it might be better to have 2 different source files. I.e. "foobar.c" and "foobar_nt.c", where the latter adds some defines and includes he former. If we stuck to this principle, we wouldn't recompile plenty of code without any changes. It add a little bit of required thought, but helps reducing compile time. I already added this principle to hal, so we wouldn't recompile the entire hal for every different configuration. I would eventuallly like to add it to the kernel as well. When 99& of the code i exactly the same, it's better to have a few extra wrapper files that include other files, so you don't have to recompile the whole thing.